test/linalg/dense.jl

```diff
@@ -582,6 +582,25 @@ end
     end
 end
 
+@testset "issue #23366 (Int Matrix to Int power)" begin
+    @testset "Tests for $elty" for elty in (Int128, Int16, Int32, Int64, Int8,
+                                            UInt128, UInt16, UInt32, UInt64, UInt8,
+                                            BigInt)
+        @test_throws DomainError elty[1 1;1 0]^-2
+        @test elty[1 1;1 0]^2 == elty[2 1;1 1]
+        I_ = elty[1 0;0 1]
+        @test I_^-1 == I_
+        if !(elty<:Unsigned)
+            @test (-I_)^-1 == -I_
+            @test (-I_)^-2 == I_
+        end
+        for elty2 = (Int64, BigInt)
+            TT = Base.promote_op(^, elty, elty2)
+            @test Base.return_types(^, (Array{elty,2}, elty2)) == [Array{TT,2}]
```

Comment:
I think the test would only check for one particular value, whereas return_type proves type correctness for all possible values of a type. And `[1 1;1 0]^big(1)` is just one example of type promotion, there's others as well, e.g. `Int32[1 1;1 0]^Int64(1)`. So it should be:
```julia
@test (@inferred elty[1 1;1 0]^elty2(1))::Matrix{TT} == [1 1;1 0]
```
I'll also add a comment to clarify.